classes/npm.bbclass

```diff
@@ -159,8 +159,10 @@ python fetch_npm() {
     with open("package.json") as infile:
         json_objs = json.load(infile)
 
-    deps = [d for d in json_objs['dependencies']]
-    json_objs.update({'bundledDependencies': deps})
+    deps = {}
+    if 'dependencies' in json_objs.keys():
+        deps = [d for d in json_objs['dependencies']]
+        json_objs.update({'bundledDependencies': deps})
```

Comment:
How about `dependencies = json_objs.get('dependencies')`? Then you could check that for None and also reuse it later.

Reply:
right. modified with suggested method 
```
deps = json_objs.get('dependencies')
if deps:
  json_objs.update({'bundledDependencies': deps})
```